x-pack/filebeat/input/netflow/decoder/v9/v9_test.go

```diff
@@ -249,3 +249,59 @@ func TestCustomFields(t *testing.T) {
 	assert.Contains(t, flows[0].Fields, "customField")
 	assert.Equal(t, flows[0].Fields["customField"], "Hello :)")
 }
+
+func TestSharedTemplates(t *testing.T) {
+	const sourceID = 1234
+	addr := test.MakeAddress(t, "127.0.0.1:12345")
+	templatePacket := []uint16{
+		// Header
+		// Version, Count, Uptime, Ts, SeqNo, Source
+		9, 1, 11, 11, 22, 22, 33, 33, 0, 1234,
+		// Set #1 (template)
+		0, 20, /*len of set*/
+		999, 3, /*len*/
+		1, 4, // Fields
+		2, 4,
+		3, 4,
+	}
+
+	t.Run("Template sharing enabled", func(t *testing.T) {
+		sharedTemplates := true
+		key := MakeSessionKey(addr, sourceID, sharedTemplates)
+		cfg := config.Defaults()
+		cfg.WithSharedTemplates(sharedTemplates)
+		proto := New(cfg)
+		flows, err := proto.OnPacket(test.MakePacket(templatePacket), addr)
```

Comment:
I think we would want some kind of scenario that adds template data from a one address and then data from different address, and demonstrates that it produces a flow. This verifies that the template is shared.

Reply:
That makes sense, I have updated the tests accordingly.